### ISSUE TITLE
[Linux] Wrapping Steam games with mullvad-exclude breaks Steam overlay.

### ISSUE THREAD

**Issue by ottergauze** [mullvad/mullvadvpn-app#6485]
### Is it a bug?

- [X] I know this is an issue with the app, and contacting Mullvad support is not relevant.

### I have checked if others have reported this already

- [X] I have checked the issue tracker to see if others have reported similar issues.

### Current Behavior

Steam Overlay's overlay does not work in games wrapped with `mullvad-exclude`.

### Expected Behavior

Steam games, when wrapped with `mullvad-exclude` within Steam on Linux, should not prevent Steam's overlay from hooking.

### Steps to Reproduce

1. Launch Steam
2. Set `mullvad-exclude %command%` as a launch parameter on any Steam game
3. Launch Game
4. Press SHIFT+TAB
5. Steam's overlay does not show.

### Operating system version

Bazzite Holographic 40 (Kinoite) x86_64 / Fedora 40

### Mullvad VPN app version

2024.3

### Additional Information

This was originally witnessed only occurring in Team Fortress 2 (native, 64 bit), but has been tested to also occur in Deep Rock Galactic (Proton), and Momentum Mod Playtest (Proton).

**Comment by ottergauze** [comment]
Still present in 2024.8.

**Comment by Neshura87** [comment]
Still present with 2025.2

System: openSUSE Tumbleweed

**Comment by hulthe** [comment]
TL;DR: Try this launch option instead:
```
mullvad-exclude env LD_PRELOAD="$LD_PRELOAD" %command%
```

Here's an explanation:

Steam uses the `LD_PRELOAD` environment variable to inject code into the game in order to get the overlay to work. Meanwhile, `mullvad-exclude` needs to run as root using `setuid` in order to get split-tunneling to work. The `LD_PRELOAD`-trick is disabled for applications using `setuid`, because if you could inject arbitrary code into applications running as root, that would be quite bad for security 😆.

The launch option works by setting LD_PRELOAD on the `%command%` executed by `mullvad-exclude`, rather than on `mullvad-exclude` itself.

**Comment by Neshura87** [comment]
Using the launch options works perfectly, thanks for the fix 👍

**Comment by hulthe** [comment]
Nice! Happy gaming :)

**Comment by Anchor5178** [comment]
Hello, thanks for this solution!

I know that the Steam-Flatpak ist not officially supported, but maybe someone can help me getting the `mullvad-exclude` command working in the Steam-Flatak? If I set the launch option of a game to `mullvad-exclude env LD_PRELOAD="$LD_PRELOAD" %command%` it wont launch, even if I just enter `mullvad-exclude %command%`.

I am running Fedora Silverblue with Mullvad installed as layered rpm and Steam installed as Flatpak.

Steam itself will launch just fine when split-tunneling but sadly the games then will launch connected to the VPN interface and thus are not split-tunneled.

**Comment by hulthe** [comment]
@Anchor5178 Flatpak is tricky. I believe that your game doesn't launch because `mullvad-exclude` is not installed inside steams flatpak sandbox.

I have no idea why running steam through split-tunneling doesn't exclude the games, excluding flatpak apps generally works. Might be that steam flatpak does something special.
